src/app/resources/iis_scripts/local_http.ps1

```diff
@@ -4,70 +4,83 @@ param(
     }
-
-    if ($content -is [System.Byte[]]) {
-        return [System.Convert]::ToBase64String($content)
-    }
-
-    return $content.ToString()
+    return [System.Convert]::ToBase64String($content).ToString()
 }
 
 ## same order as @angular/http/enums/RequestMethod
+## Note that index 0 maps to GET, so if $reqObj.method is null we would default to GET
 $requestMethods = @(
-    [Microsoft.PowerShell.Commands.WebRequestMethod]::Get,
-    [Microsoft.PowerShell.Commands.WebRequestMethod]::Post,
-    [Microsoft.PowerShell.Commands.WebRequestMethod]::Put,
-    [Microsoft.PowerShell.Commands.WebRequestMethod]::Delete,
-    [Microsoft.PowerShell.Commands.WebRequestMethod]::Options,
-    [Microsoft.PowerShell.Commands.WebRequestMethod]::Head,
-    [Microsoft.PowerShell.Commands.WebRequestMethod]::Patch
+    [System.Net.Http.HttpMethod]::Get,
+    [System.Net.Http.HttpMethod]::Post,
+    [System.Net.Http.HttpMethod]::Put,
+    [System.Net.Http.HttpMethod]::Delete,
+    [System.Net.Http.HttpMethod]::Options,
+    [System.Net.Http.HttpMethod]::Head,
+    [System.Net.Http.HttpMethod]::Patch
 )
 
 $decoded = $contentEncoding.GetString([System.Convert]::FromBase64String($requestBase64))
 $reqObj = ConvertFrom-Json $decoded
-$uri = [System.UriBuilder]$reqObj.url
-$uri.Host = "localhost"
-$req = @{ "Uri" = $uri.ToString() }
-
-if ($reqObj.method) {
-    $req.Method = $requestMethods[$reqObj.method]
-}
 
-if ($reqObj._body) {
-    $req.Body = $reqObj._body
-}
+$httpMethod = $requestMethods[[int]$reqObj.method]
+$uriBuilder = [System.UriBuilder]$reqObj.url
+$uriBuilder.Host = "localhost"
+$uri = $uriBuilder.ToString()
 
-if ($reqObj.headers) {
-    $req.Headers = @{}
+Write-Host $decoded
+try {
+    $httpMsg = New-Object System.Net.Http.HttpRequestMessage -ArgumentList $httpMethod,$uri
+    if ($reqObj.Body) {
+        $httpMsg.Content = $requestContent
+        $requestContent = New-Object System.Net.Http.StringContent ([string] $reqObj.Body)
+    }
     foreach ($prop in $reqObj.headers | Get-Member -MemberType NoteProperty | Select-Object -ExpandProperty Name) {
-        ## NOTE: not all http headers follow the same syntax. This delimitation logic is assumed to work for all headers we use
-        $req.Headers[$prop] = $reqObj.headers.$prop -Join ","
+        $httpMsg.Headers.Add($prop, $reqObj.headers.$prop)
     }
-}
+    $clientHandler = New-Object System.Net.Http.HttpClientHandler
+    $clientHandler.UseDefaultCredentials = $true
+    $client = New-Object System.Net.Http.HttpClient -ArgumentList $clientHandler
+    $responseMsg = $client.SendAsync($httpMsg).GetAwaiter().GetResult()
 
-try {
-    $res = Invoke-WebRequest -UseBasicParsing -UseDefaultCredentials @req
-} catch {
-    $errMsg = $_.Exception.Message
-    $res = $_.Exception.Response
-} finally {
-    if (!$res) {
-        throw $errMsg
+    if ($responseMsg.Content) {
+        $resContent = stringify $responseMsg.Content.ReadAsByteArrayAsync().GetAwaiter().GetResult()

```

Comment:
declaration of $resContent is missing. Please put $resContent = $null before the if statement.

Reply:
Okay, I did some testing and missing declaration is not that harmful in Powershell script. We can revisit this later.